framework/src/outputs/Output.C

```diff
@@ -174,6 +183,14 @@ Output::Output(const InputParameters & parameters)
     for (auto i = 0; i < pwb_olf->functionSize(); i++)
       _sync_times.insert(pwb_olf->domain(i));
   }
+
+  // Check that the sync times were retrieved as expected
+  if (isParamSetByUser("sync_times_object") && !_sync_times_object)
+    paramError("sync_times_object", "Times object does not exist");
```

Comment:
should be able to remove this as if casting fails it will fail in the `getUserObject` call. But if you want to be doubly safe and choose to keep this, then ignore my `static_cast` suggestion above